shared/dataflow/codeql/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -1691,6 +1691,55 @@ module Make<DF::InputSig DataFlowLang, InputSig<DataFlowLang> Input> {
             interpretInput(input, input.getNumToken(), ref, node)
           )
         }
+
+        /** Provides query predicates for outputting a set of relevant sources and sinks. */
+        module TestSourceSinkOutput {
+          final private class SourceOrSinkElementFinal = SourceOrSinkElement;
+
+          abstract class RelevantSourceCallable extends SourceOrSinkElementFinal {
```

Comment:
I think instead of repeating this bad pattern from `TestOutput`, we should make it a parameter of `TestSourceSinkOutput` instead.

Reply:
And ideally also update `TestOutput` to use a module parameter instead of an abstract class.